feat(Chip): Add read-only, icon, badge variants; deprecate status label

jmuzina opened the pull request:
## Done

- Deprecates the Status Label component.
  - Updates usages of the status label component to use the readonly chip. 
    - In examples
    - On the Vanilla docs site
- Adds the following variants to the Chip component
  - Read-only (subsumes the Status Label component)
  - With icon
  - With badge

Fixes #5535 
Fixes [WD-22497](https://warthogs.atlassian.net/browse/WD-22497)

## TODO

- [x] Add design guidelines per [design docs](https://docs.google.com/document/d/1HBZ_i7nI0FQbR9FJ_SE0x3Ax_BUi4DnBXMEhZPp2BiI/edit?tab=t.5qfuxdsojs30)
- [x] Add accessibility guidelines
- [x] Fix excess whitespace introduced by the example helper macro

## QA

- Open [chip examples](https://vanilla-framework-5555.demos.haus/docs/examples/patterns/chip/combined?theme=light) and verify they appear as expected
- Review [chip docs](https://vanilla-framework-5555.demos.haus/docs/patterns/chip)
  - Make sure to check the "Design guidelines" and "Accessibility" (WIP) tabs, as these have been updated / added.
  - Disregard excess whitespace in the chip markup - this will be addressed prior to final QA review
- Review [status label docs](https://vanilla-framework-5555.demos.haus/docs/patterns/status-labels) and verify they have been marked as deprecated properly, with a link to the read-only chip.
- Review [4.25.0 release notes](https://vanilla-framework-5555.demos.haus/docs/whats-new)
- Browse around the Vanilla website and verify that the status label has been replaced with the readonly chip in all non-deprecated pages.
  - Sitewide side navigation, on recently updated components
  - [What's New](https://vanilla-framework-5555.demos.haus/docs/whats-new)
  - In [variant headings](https://vanilla-framework-5555.demos.haus/docs/patterns/chip#read-only-new)

### Check if PR is ready for release

If this PR contains Vanilla SCSS or macro code changes, it should contain the following changes to make sure it's ready for the release:

- [ ] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [ ] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical/vanilla-framework/releases/latest), following semver convention
  - if existing APIs (CSS classes & macro APIs) are not changed it can be a bugfix release (x.x.**X**)
  - if existing APIs (CSS classes & macro APIs) are changed/added/removed it should be a minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [ ] Any changes to component class names (new patterns, variants, removed or added features) or macros should be listed on the [what's new page](https://github.com/canonical/vanilla-framework/blob/main/releases.yml).


## Screenshots

![Screenshot from 2025-06-23 17-58-23](https://github.com/user-attachments/assets/8a97876e-38c1-4c0d-8f5a-829be6021097)
![Screenshot from 2025-06-23 17-58-26](https://github.com/user-attachments/assets/3090bd45-c9a0-408a-a919-b51afc990f31)
![Screenshot from 2025-06-23 17-58-30](https://github.com/user-attachments/assets/6e23e21f-14b1-4927-bae1-fd0c100dfae0)
![Screenshot from 2025-06-23 17-58-34](https://github.com/user-attachments/assets/0d6b77fd-77b2-4777-a228-882662fe2d40)
![Screenshot from 2025-06-23 17-58-38](https://github.com/user-attachments/assets/31653bda-f3c6-4120-b03a-9379b53f1ef4)
![Screenshot from 2025-06-23 17-58-43](https://github.com/user-attachments/assets/3ac9a672-6353-455e-a740-1264003b1c53)



[WD-22497]: https://warthogs.atlassian.net/browse/WD-22497?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

webteam-app commented:
### [<img src='https://assets.ubuntu.com/v1/6baef514-ubuntu-circle-of-friends-large.svg' height=32 width=32> Demo</img>](https://vanilla-framework-5555.demos.haus)
### [<img src='https://assets.ubuntu.com/v1/e512b0e2-jenkins.svg' height=32 width=32> Jenkins </img>](https://jenkins.canonical.com/webteam/job/start-demo/7723)
### [<img src='https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg' height=32 width=32> demos.haus </img>](https://demos.haus)

jmuzina commented:
Added a [draft](https://docs.google.com/document/d/1HBZ_i7nI0FQbR9FJ_SE0x3Ax_BUi4DnBXMEhZPp2BiI/edit?tab=t.ujmt11wb5pbr) for how we might lay out the accessibility docs. 

Also added a [draft](https://docs.google.com/document/d/1HBZ_i7nI0FQbR9FJ_SE0x3Ax_BUi4DnBXMEhZPp2BiI/edit?tab=t.5qfuxdsojs30#heading=h.6mc1so55f0as) for the tokens docs section. 

@timea-udvarhelyi

jmuzina commented:
Opening for design review.

There are still some updates to the Design docs pending. 

- [ ] @timea-udvarhelyi to check on whether we should deploy the site with the images that are currently in use for the "do/don't" examples
- [ ] @timea-udvarhelyi to double check if the colors used in the component anatomy screenshot are approved for publishing
- [ ] @timea-udvarhelyi To make some adjustments to the "Styling" section of the docs, consolidating layout and spacing. 

@timea-udvarhelyi let me know if these are accomplishable soon - I'd like to merge and release this tomorrow ideally. If needed we can deliver the documentation separately from the chip updates.

timea-udvarhelyi commented:
@jmuzina I updated the anatomy image and checked the examples, there are permitted to be published except for one that I have changed in the doc.

The correct colors are used in the Figma component as tokens.

The MVP for the Styling and Accessibility sections are ready. I would like your help reviewing the spacing color and responsiveness sections. But this should not be a blocker.

The anatomy section has also been updated to contain the mention of the dismissible icon subcomponent.

I reviewed the icon radius change, it looks good to me! +1 from visual review

jmuzina commented:
@timea-udvarhelyi I am leaving out one requested change (not to the Chip, but to the Vanilla sidenav to use the readonly chip instead of the status label.

The Figma example of this shows the chip at the end of the navigation item. However, the side navigation item elements are not currently set up to support this - when i tried to implement it the chips just appeared inline next to the nav item.
![image](https://github.com/user-attachments/assets/4ffc6e79-40e6-40fa-951a-c8f2b79b6ce6)

This seems to require a change to the side navigation component, perhaps to make the side navigation item components flexboxes that place the link text at the left and the status chip (if present) at the right. This is enough of a change that I'm going to hold off on merging this PR and we can discuss it next week.

jmuzina commented:
Noticed a few [visual regressions](https://percy.io/bb49709b/web/vanilla-framework/builds/41251809?utm_campaign=bb49709b&utm_content=vanilla-framework&utm_source=github_status_private) that I'll look to fix on Monday.

I've also made some changes to make the read-only chip compatible with being used in p-side-navigation__status, and replaced usages of p-status-label across the site pages and non-deprecated examples with p-chip.is-readonly.

QA instructions are updated with links to check out if you're curious how that looks.

I also need to update the docs a bit to match Timea's new sections.

jmuzina commented:
@timea-udvarhelyi while applying the read-only chip to the Vanilla website, I noticed an interesting point we should probably hash out here:

Currently, chips have `20px` bottom margin and `8px` right margin, to space them away from content that follows (i.e, placing multiple chips next to each other, such as in Search & Filter)

![image](https://github.com/user-attachments/assets/a3f92742-0bb3-4028-bf36-455b6f925ac3)

If the read-only chip had the same margin, it would cause unexpected spacing in contexts where it is inline (such as navigation or tables)

![image](https://github.com/user-attachments/assets/f384081c-be62-4261-a1e7-66e7eb097ead)
![image](https://github.com/user-attachments/assets/79e30013-1e74-438b-b5a1-e4e905a784e8)

So, I've set the read-only chips to have no margin:

![image](https://github.com/user-attachments/assets/9401aa0a-95e5-4b47-9e57-42b984c1f2f4)
![image](https://github.com/user-attachments/assets/4925f040-a886-4e31-8247-1a1e9f13aa3f)


However, this means that a cluster of read-only chips in succession would be very close together. 

![image](https://github.com/user-attachments/assets/04e900b6-cb23-4113-befa-5a2ce671e144)

Would multiple read-only chips ever be used in the same block?
